runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -991,6 +1015,46 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 			? getDataAddrForDiscontiguous(arrayPtr)
 			: getDataAddrForContiguous(arrayPtr);
 	}
+
+	/**
+	 * Asserts that the dataAddr field of the indexable object is correct.
+	 *
+	 * @param arrayPtr      Pointer to the indexable object
+	 */
+	MMINLINE bool
+	isCorrectDataAddr(J9IndexableObject *arrayPtr)
+	{
+		ArrayLayout layout = getArrayLayout(arrayPtr);
+		bool isDiscontiguous = (layout != InlineContiguous);
+
+		return isDiscontiguous
+			? isCorrectDataAddrForDiscontiguousArraylet(arrayPtr)
+			: isCorrectDataAddrForContiguousArraylet(arrayPtr);
```

Comment:
Why the reverse logic? Why not:
```
	return (InlineContiguous == layout)
		? isCorrectDataAddrForContiguousArraylet(arrayPtr)
		: isCorrectDataAddrForDiscontiguousArraylet(arrayPtr);
```

Reply:
I used the formatting of the method above for consistency, even though I didn't like it. I fully agree this logic is better than the reverse logic and will gladly change it.

Reply:
Are you going to change it as suggested?